File: KTCK_QuanLySinhVien/TrangChu.vb
Public Class TrangChu

    Private _dbAccess As New DataBaseAccess
    Private _isLoading As Boolean = False
    Public Sub loadData()
        loadDataOnGridView()
        loadDataOnCombobox()
    End Sub

    Public Sub loadDataOnCombobox()
        Dim sql As String = "Select * from dbo.Lop"
        Dim dTable As DataTable = _dbAccess.GetDataTable(sql)
        Me.Class_cbb.DataSource = dTable
        Me.Class_cbb.ValueMember = "TenLop"
        Me.Class_cbb.DisplayMember = "TenLop"
    End Sub

    Private Sub loadDataOnGridView(tenLop As String)
        Dim sqlQuery As String = String.Format("Select * from SinhVien where Lop = '" & tenLop & "'")
        Dim dTable As DataTable = _dbAccess.GetDataTable(sqlQuery)
        Me.dgvStudents.DataSource = dTable
        With Me.dgvStudents
            .Columns(0).HeaderText = "Mã sinh viên"
            .Columns(1).HeaderText = "Tên"
            .Columns(1).Width = 200
            .Columns(2).HeaderText = "Lớp"
            .Columns(3).HeaderText = "Giới tính"
            .Columns(4).HeaderText = "Số điện thoại"
            .Columns(5).HeaderText = "Email"
            .Columns(5).Width = 200
        End With
    End Sub
    Public Sub loadDataOnGridView()
        Dim sqlQuery As String = String.Format("Select * from SinhVien ")
        Dim dTable As DataTable = _dbAccess.GetDataTable(sqlQuery)
        Me.dgvStudents.DataSource = dTable
        With Me.dgvStudents
            .Columns(0).HeaderText = "Mã sinh viên"
            .Columns(1).HeaderText = "Tên"
            .Columns(1).Width = 200
            .Columns(2).HeaderText = "Lớp"
            .Columns(3).HeaderText = "Giới tính"
            .Columns(4).HeaderText = "Số điện thoại"
            .Columns(5).HeaderText = "Email"
            .Columns(5).Width = 200
        End With
    End Sub

    Private Sub TrangChu_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        _isLoading = True
        loadDataOnCombobox()
        loadDataOnGridView()
        Class_cbb.Text = ""
        txt_timKiem.Enabled = False
        _isLoading = False
    End Sub



    Private Sub Delete_btn_Click(sender As Object, e As EventArgs) Handles Delete_btn.Click
        If MsgBox("Bạn có muốn xóa không !", vbYesNo, "Xác nhận") = 6 Then
            Xoa()

        End If
    End Sub
    Private Sub Xoa()

        'Khai bao bien lay StudentID ma dong can xoa da duoc chon tren gridview
        Dim MSV As String = Me.dgvStudents.Rows(Me.dgvStudents.CurrentCell.RowIndex).Cells("MSV").Value
        'Khai bao cau lenh Query de xoa
        Dim sqlQuery As String = String.Format("DELETE from SinhVien WHERE MSV='{0}'", MSV)
        'Thuc hien xoa
        If _dbAccess.ExecuteNoneQuery(sqlQuery) Then 'Xoa thanh cong thi thong bao
            MessageBox.Show("Xóa thành công!”)
            'Load lai du lieu tren Gridview
            loadDataOnGridView()
        Else
            MessageBox.Show("Xóa thất bại!")
        End If
    End Sub

    Private Sub them_btn_Click(sender As Object, e As EventArgs) Handles them_btn.Click
        Them.ShowDialog()


    End Sub

    Private Sub update_btn_Click(sender As Object, e As EventArgs) Handles update_btn.Click
        Dim editForm As New Sua
        editForm.msv_txt.Text = Me.dgvStudents.Rows(Me.dgvStudents.CurrentCell.RowIndex).Cells("MSV").Value
        With Me.dgvStudents
            editForm.name_TextBox.Text = .Rows(.CurrentCell.RowIndex).Cells("MSV").Value
            editForm.name_TextBox.Text = .Rows(.CurrentCell.RowIndex).Cells("Ten").Value
            editForm.class_cbb.Text = .Rows(.CurrentCell.RowIndex).Cells("Lop").Value
            editForm.GenderComboBox.Text = .Rows(.CurrentCell.RowIndex).Cells("GioiTinh").Value
            editForm.phone_TextBox.Text = .Rows(.CurrentCell.RowIndex).Cells("SDT").Value
            editForm.email_TextBox.Text = .Rows(.CurrentCell.RowIndex).Cells("Email").Value
        End With
        editForm.ShowDialog()
    End Sub



    Private Sub Class_cbb_SelectedIndexChanged(sender As Object, e As EventArgs) Handles Class_cbb.SelectedIndexChanged
        If Not _isLoading Then
            If txt_timKiem.Text = "" Then
                loadDataOnGridView(Me.Class_cbb.SelectedValue)
            Else
                timKiem(Me.cbb_timKiem.Text, txt_timKiem.Text, Class_cbb.Text)
            End If

        End If
    End Sub

    Private Sub lamMoi_btn_Click(sender As Object, e As EventArgs) Handles lamMoi_btn.Click
        _isLoading = True
        loadDataOnCombobox()
        loadDataOnGridView()
        Class_cbb.Text = ""
        txt_timKiem.Text = ""
        cbb_timKiem.Text = ""
        txt_timKiem.Enabled = False
        _isLoading = False
    End Sub

    Private Sub txt_timKiem_TextChanged(sender As Object, e As EventArgs) Handles txt_timKiem.TextChanged
        If Class_cbb.Text = "" Then
            timKiem(Me.cbb_timKiem.Text, txt_timKiem.Text)
        Else
            timKiem(Me.cbb_timKiem.Text, txt_timKiem.Text, Class_cbb.Text)
        End If
    End Sub

    Private Sub timKiem(column As String, value As String)
        Dim sqlQuery As String = String.Format("Select * from SinhVien where " & column & " like N'%" & value & "%'")
        Dim dTable As DataTable = _dbAccess.GetDataTable(sqlQuery)
        Me.dgvStudents.DataSource = dTable
        With Me.dgvStudents
            .Columns(0).HeaderText = "Mã sinh viên"
            .Columns(1).HeaderText = "Tên"
            .Columns(1).Width = 200
            .Columns(2).HeaderText = "Lớp"
            .Columns(3).HeaderText = "Giới tính"
            .Columns(4).HeaderText = "Số điện thoại"
            .Columns(5).HeaderText = "Email"
            .Columns(5).Width = 200
        End With
    End Sub

    Private Sub timKiem(column As String, value As String, lop As String)
        Dim sqlQuery As String = String.Format("Select * from SinhVien where " & column & " like N'%" & value & "%' and Lop = '" & lop & "'")
        Dim dTable As DataTable = _dbAccess.GetDataTable(sqlQuery)
        Me.dgvStudents.DataSource = dTable
        With Me.dgvStudents
            .Columns(0).HeaderText = "Mã sinh viên"
            .Columns(1).HeaderText = "Tên"
            .Columns(1).Width = 200
            .Columns(2).HeaderText = "Lớp"
            .Columns(3).HeaderText = "Giới tính"
            .Columns(4).HeaderText = "Số điện thoại"
            .Columns(5).HeaderText = "Email"
            .Columns(5).Width = 200
        End With
    End Sub

    Private Sub cbb_timKiem_SelectedIndexChanged(sender As Object, e As EventArgs) Handles cbb_timKiem.SelectedIndexChanged
        txt_timKiem.Enabled = True
    End Sub

    Private Sub tkb_btn_Click(sender As Object, e As EventArgs) Handles tkb_btn.Click
        Dim tkbForm As New ThoiKhoaBieu
        tkbForm.msv_label.Text = Me.dgvStudents.Rows(Me.dgvStudents.CurrentCell.RowIndex).Cells("MSV").Value
        tkbForm.tenSV_label.Text = Me.dgvStudents.Rows(Me.dgvStudents.CurrentCell.RowIndex).Cells("Ten").Value
        tkbForm.ShowDialog()
    End Sub
End Class

File: KTCK_QuanLySinhVien/View/TrangChu.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace KTCK_QuanLySinhVien
{
    public partial class TrangChu
    {


        private DataBaseAccess _dbAccess = new DataBaseAccess();
        private bool _isLoading = false;
        

        public TrangChu()
        {
            InitializeComponent();
        }
        public void loadData()
        {
            loadDataOnGridView();
            loadDataOnCombobox();
        }

        public void loadDataOnCombobox()
        {
            string sql = "Select * from dbo.Lop";
            var dTable = _dbAccess.GetDataTable(sql);
            Class_cbb.DataSource = dTable;
            Class_cbb.ValueMember = "TenLop";
            Class_cbb.DisplayMember = "TenLop";
        }

        private void loadDataOnGridView(string tenLop)
        {
            string sqlQuery = string.Format("Select * from SinhVien where Lop = '" + tenLop + "'");
            var dTable = _dbAccess.GetDataTable(sqlQuery);
            dgvStudents.DataSource = dTable;
            {
                var withBlock = dgvStudents;
                withBlock.Columns[0].HeaderText = "Mã sinh viên";
                withBlock.Columns[1].HeaderText = "Tên";
                withBlock.Columns[1].Width = 200;
                withBlock.Columns[2].HeaderText = "Lớp";
                withBlock.Columns[3].HeaderText = "Giới tính";
                withBlock.Columns[4].HeaderText = "Số điện thoại";
                withBlock.Columns[5].HeaderText = "Email";
                withBlock.Columns[5].Width = 200;
            }
        }
        public void loadDataOnGridView()
        {
            string sqlQuery = string.Format("Select * from SinhVien ");
            var dTable = _dbAccess.GetDataTable(sqlQuery);
            dgvStudents.DataSource = dTable;
            {
                var withBlock = dgvStudents;
                withBlock.Columns[0].HeaderText = "Mã sinh viên";
                withBlock.Columns[1].HeaderText = "Tên";
                withBlock.Columns[1].Width = 200;
                withBlock.Columns[2].HeaderText = "Lớp";
                withBlock.Columns[3].HeaderText = "Giới tính";
                withBlock.Columns[4].HeaderText = "Số điện thoại";
                withBlock.Columns[5].HeaderText = "Email";
                withBlock.Columns[5].Width = 200;
            }
        }

        private void TrangChu_Load(object sender, EventArgs e)
        {
            _isLoading = true;
            _dbAccess.CreateXmlFile("SinhVien");
            loadDataOnCombobox();
            loadDataOnGridView();
            Class_cbb.Text = "";
            txt_timKiem.Enabled = false;
            _isLoading = false;
        }



        private void Delete_btn_Click(object sender, EventArgs e)
        {
            if ((int)Interaction.MsgBox("Bạn có muốn xóa không !", Constants.vbYesNo, "Xác nhận") == 6)
            {
                Xoa();

            }
        }
        private void Xoa()
        {

            // Khai bao bien lay StudentID ma dong can xoa da duoc chon tren gridview
            string MSV = Conversions.ToString(dgvStudents.Rows[dgvStudents.CurrentCell.RowIndex].Cells["MSV"].Value);
            // Khai bao cau lenh Query de xoa
            string sqlQuery = string.Format("DELETE from SinhVien WHERE MSV='{0}'", MSV);
            // Thuc hien xoa
            if (_dbAccess.ExecuteNoneQuery(sqlQuery)) // Xoa thanh cong thi thong bao
            {
                MessageBox.Show("Xóa thành công!");
                // Load lai du lieu tren Gridview
                loadDataOnGridView();
            }
            else
            {
                MessageBox.Show("Xóa thất bại!");
            }
        }

        private void them_btn_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.Them.ShowDialog();


        }

        private void update_btn_Click(object sender, EventArgs e)
        {
            var editForm = new Sua();
            editForm.msv_txt.Text = Conversions.ToString(dgvStudents.Rows[dgvStudents.CurrentCell.RowIndex].Cells["MSV"].Value);
            {
                var withBlock = dgvStudents;
                editForm.name_TextBox.Text = Conversions.ToString(withBlock.Rows[withBlock.CurrentCell.RowIndex].Cells["MSV"].Value);
                editForm.name_TextBox.Text = Conversions.ToString(withBlock.Rows[withBlock.CurrentCell.RowIndex].Cells["Ten"].Value);
                editForm.class_cbb.Text = Conversions.ToString(withBlock.Rows[withBlock.CurrentCell.RowIndex].Cells["Lop"].Value);
                editForm.GenderComboBox.Text = Conversions.ToString(withBlock.Rows[withBlock.CurrentCell.RowIndex].Cells["GioiTinh"].Value);
                editForm.phone_TextBox.Text = Conversions.ToString(withBlock.Rows[withBlock.CurrentCell.RowIndex].Cells["SDT"].Value);
                editForm.email_TextBox.Text = Conversions.ToString(withBlock.Rows[withBlock.CurrentCell.RowIndex].Cells["Email"].Value);
            }
            editForm.ShowDialog();
        }



        private void Class_cbb_SelectedIndexChanged(object sender, EventArgs e)
        {
            if (!_isLoading)
            {
                if (string.IsNullOrEmpty(txt_timKiem.Text))
                {
                    loadDataOnGridView(Conversions.ToString(Class_cbb.SelectedValue));
                }
                else
                {
                    timKiem(cbb_timKiem.Text, txt_timKiem.Text, Class_cbb.Text);
                }

            }
        }

        private void lamMoi_btn_Click(object sender, EventArgs e)
        {
            _isLoading = true;
            loadDataOnCombobox();
            loadDataOnGridView();
            Class_cbb.Text = "";
            txt_timKiem.Text = "";
            cbb_timKiem.Text = "";
            txt_timKiem.Enabled = false;
            _isLoading = false;
        }

        private void txt_timKiem_TextChanged(object sender, EventArgs e)
        {
            if (string.IsNullOrEmpty(Class_cbb.Text))
            {
                timKiem(cbb_timKiem.Text, txt_timKiem.Text);
            }
            else
            {
                timKiem(cbb_timKiem.Text, txt_timKiem.Text, Class_cbb.Text);
            }
        }

        private void timKiem(string column, string value)
        {
            string sqlQuery = string.Format("Select * from SinhVien where " + column + " like N'%" + value + "%'");
            var dTable = _dbAccess.GetDataTable(sqlQuery);
            dgvStudents.DataSource = dTable;
            {
                var withBlock = dgvStudents;
                withBlock.Columns[0].HeaderText = "Mã sinh viên";
                withBlock.Columns[1].HeaderText = "Tên";
                withBlock.Columns[1].Width = 200;
                withBlock.Columns[2].HeaderText = "Lớp";
                withBlock.Columns[3].HeaderText = "Giới tính";
                withBlock.Columns[4].HeaderText = "Số điện thoại";
                withBlock.Columns[5].HeaderText = "Email";
                withBlock.Columns[5].Width = 200;
            }
        }

        private void timKiem(string column, string value, string lop)
        {
            string sqlQuery = string.Format("Select * from SinhVien where " + column + " like N'%" + value + "%' and Lop = '" + lop + "'");
            var dTable = _dbAccess.GetDataTable(sqlQuery);
            dgvStudents.DataSource = dTable;
            {
                var withBlock = dgvStudents;
                withBlock.Columns[0].HeaderText = "Mã sinh viên";
                withBlock.Columns[1].HeaderText = "Tên";
                withBlock.Columns[1].Width = 200;
                withBlock.Columns[2].HeaderText = "Lớp";
                withBlock.Columns[3].HeaderText = "Giới tính";
                withBlock.Columns[4].HeaderText = "Số điện thoại";
                withBlock.Columns[5].HeaderText = "Email";
                withBlock.Columns[5].Width = 200;
            }
        }

        private void cbb_timKiem_SelectedIndexChanged(object sender, EventArgs e)
        {
            txt_timKiem.Enabled = true;
        }

        private void tkb_btn_Click(object sender, EventArgs e)
        {
            var tkbForm = new ThoiKhoaBieu();
            tkbForm.msv_label.Text = Conversions.ToString(dgvStudents.Rows[dgvStudents.CurrentCell.RowIndex].Cells["MSV"].Value);
            tkbForm.tenSV_label.Text = Conversions.ToString(dgvStudents.Rows[dgvStudents.CurrentCell.RowIndex].Cells["Ten"].Value);
            tkbForm.ShowDialog();
        }


    }
}